src/main/java/dev/openfeature/sdk/FeatureProvider.java

```diff
@@ -22,4 +22,29 @@ default List<Hook> getProviderHooks() {
     ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue, EvaluationContext ctx);
 
     ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue, EvaluationContext ctx);
+
+    /**
+     * This method is called before a provider is used to evaluate flags. Providers can overwrite this method,
+     * if they have special initialization needed prior being called for flag evaluation.
+     * <p>
+     * It is ok, if the method is expensive as it is executed in the background. All runtime exceptions will be
+     * caught and logged.
+     * </p>
+     */
+    default void initialize() {
+        // Intentionally left blank
+    }
+
+    /**
+     * This method is called when a new provider is about to be used to evaluate flags.
```

Comment:
```suggestion
     * This method is called when a new provider is about to be used to evaluate flags, or the SDK is shut down.
```

Reply:
Related: https://github.com/open-feature/java-sdk/pull/456#issuecomment-1568938404

Reply:
done

Reply:
i think, my reasoning not to publish it was to not implement yet another requirement in the PR. but to be frank, i can't remember... 🤣